### ISSUE TITLE
SQLAnywhere Driver segfaults on Mac OS X

### ISSUE THREAD

**Issue by rwaldvogel** [sqlanywhere/sqlanywhere#1]
I am trying to connect to a remote SQLA10 database running on Windows box with the  sqlanywhere driver.

Installing it works fine (gem install sqlanywhere -> no errors).

When trying the example, I get the following segfault:

ruby-1.8.7-p302 > require 'rubygems'
 => true 
ruby-1.8.7-p302 > require 'sqlanywhere'
 => true 
ruby-1.8.7-p302 > api = SQLAnywhere::SQLAnywhereInterface.new()
 => #SQLAnywhere::SQLAnywhereInterface:0x10051c380 
ruby-1.8.7-p302 > SQLAnywhere::API.sqlany_initialize_interface( api )
 => 0 
ruby-1.8.7-p302 > 
ruby-1.8.7-p302 >   api.sqlany_init()
(irb):8: [BUG] Segmentation fault
ruby 1.8.7 (2010-08-16 patchlevel 302) [i686-darwin10.5.0]

Abort trap

**Comment by sqlanywhere** [comment]
What version of SQL Anywhere 10 are you using? You can find out by running:

dbsrv10 -v

Thanks,
- Eric

**Comment by rwaldvogel** [comment]
On the client side I have SQLA installed all together.
Debugging a little bit, I found, that you rely on the Sybase client library in some sort. 
If this is correct, my problem is clear: No client, no initialize.
But I would like to ask you to change the requirements: SQLA Client installed.
Reading your requirements (no requirements at all), I concluded that you in some sort did a TDS implementation, meaning you don't have any dependency to Sybase installed. I really would appreciate if your dependency would be stated clearly. Otherwise why not develop a real independent SQLA driver?

Thanks 
Ralf

**Comment by sqlanywhere** [comment]
Hello Ralf,

As stated in the README file, "This library wraps the functionality provided by the SQL Anywhere DBCAPI library." DBCAPI is the SQL Anywhere C API runtime (libdblib10_r.dylib on Mac OSX installs). You must have this file, along with all the other files required for a client install in order to be able to connect ( http://dcx.sybase.com/index.html#html/dbpgen10/pg-programming-generatedid-1507.html ).

(I understand this will not help you since you are on SQLA10, but for other people reading this, Sybase recently released a pre-built Client Install for SQLA12 that can be found at http://www.sybase.com/detail?id=1087327 )

The SQL Anywhere C API runtime (DBCAPI) was introduced after SQLA10 was released. The SQL Anywhere C API runtime libraries were added to SQLA10 in the 10.0.1.3767 EBF. You will need to update to at least this version in order to have access to the SQL Anywhere C API runtime libraries. 

This is not a TDS implementation. It uses the standard Embedded SQL interfaces (which is why it requires the SQL Anywhere client libraries). At this time, there are no plans to develop a pure ruby driver (that is, a driver that requires no client components). 

In summary, the steps are:
1. Make sure your SQL Anywhere installation is 10.0.1.3767 or greater
2. Make sure all of the client files are available on the machine you are trying to connect from.
3. Make sure that the libraries (including libdblib10_r.dylib) are in your library path. For more information on sourcing the files, see http://dcx.sybase.com/index.html#html/dbdaen10/da-dyld-library-path-environment-variable.html .

I will update the sqlanywhere ruby driver documentation with this required information. Thank you for your feedback.
- Eric

**Comment by rwaldvogel** [comment]
Hello Eric,

Thanks for updating the readme file and making it as clear possible. 

Ralf
